crates/re_types/src/archetypes/text_document_ext.rs

```diff
@@ -0,0 +1,35 @@
+use crate::components::MediaType;
+
+use super::TextDocument;
+
+impl TextDocument {
+    /// Creates a new [`TextDocument`] from a utf8 file.
+    ///
+    /// The media type will be inferred from the path (extension), or the contents if that fails.
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn from_file(filepath: impl AsRef<std::path::Path>) -> anyhow::Result<Self> {
+        use anyhow::Context as _;
+
+        let filepath = filepath.as_ref();
+        let contents = std::fs::read(filepath)
+            .with_context(|| format!("could not read file contents: {filepath:?}"))?;
+        Self::from_file_contents(contents, MediaType::guess_from_path(filepath))
+            .with_context(|| format!("could not parse file contents: {filepath:?}"))
+    }
+
+    /// Creates a new [`TextDocument`] from the contents of a utf8 file.
+    ///
+    /// If unspecified, the media type will be inferred from the contents.
+    #[inline]
+    pub fn from_file_contents(
+        contents: Vec<u8>,
+        media_type: Option<impl Into<MediaType>>,
+    ) -> anyhow::Result<Self> {
+        let media_type = media_type.map(Into::into);
+        let media_type = MediaType::or_guess_from_data(media_type, &contents);
```

Comment:
What text types can be guessed from data? O.o

Reply:
```sh
$ head whatami
<!DOCTYPE html>
<!-- The version of index.html that will be served by app.rerun.io -->
<html>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />

  <!-- Disable zooming: -->
  <meta
    name="viewport"
    content="width=device-width, initial-scale=1.0, user-scalable=no"
  />

$ file --mime-type whatami
./whatami: text/html
```
:shrug: